2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors/scalastream/CollectorService.scala

```diff
@@ -81,22 +84,97 @@ class CollectorService(
           optionalHeaderValueByName("User-Agent") { userAgent =>
             optionalHeaderValueByName("Referer") { refererURI =>
               headerValueByName("Raw-Request-URI") { rawRequest =>
-                hostName { host =>
-                  clientIP { ip =>
-                    requestInstance{ request =>
-                      entity(as[String]) { body =>
+                headerValueByName("Kinesis-Part-Key") { kinesisPartKey =>
```

Comment:
Combining them will require me to make significant changes to the automated tests, I'm thinking that I'd rather keep them separated out for MVP purposes. I just tested locally (and soon will test in the dev. env), and everything seems to be working well.